src/infer_response.cc

```diff
@@ -282,6 +305,24 @@ InferenceResponse::TraceOutputTensors(
 }
 #endif  // TRITON_ENABLE_TRACING
 
+#ifdef TRITON_ENABLE_METRICS
+void
+InferenceResponse::UpdateResponseMetrics() const
+{
+  if (model_ != nullptr && seq_num_ == 0) {
```

Comment:
General question - the response factory already has this function and a notion of which response is being processed (response stats index):
```cpp
  uint64_t GetAndIncrementResponseIndex() { return response_stats_index_++; };
```

Can this be used instead of the new logic you have added for tracking the response count?

Reply:
General context on the feature: https://github.com/triton-inference-server/core/pull/325

Reply:
I couldn't find any places where`GetAndIncrementResponseIndex` is called except in square backend. @kthui should have some better ideas.
https://github.com/search?q=org%3Atriton-inference-server+TRITONBACKEND_ModelInstanceReportResponseStatistics&type=code

Reply:
If I understand correctly
```c++
std::atomic<uint64_t> response_cnt_;
```
means the same as
```c++
std::atomic<uint64_t> response_stats_index_;
```
?
In this case, they should be one variable.

Reply:
Then it makes sense to "reuse"
```c++
uint64_t GetAndIncrementResponseIndex() { return response_stats_index_++; };
```
for the same purpose, but some renaming may be needed.

Reply:
Reused `response_stats_index_` and renamed to `response_index_` in the latest commit. @rmccorm4

Reply:
I take that back. It might affect Square Backend.